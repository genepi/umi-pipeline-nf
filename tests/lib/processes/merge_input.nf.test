nextflow_process {

    name "Test Process MERGE_FASTQ"
    script "lib/processes/merge_input.nf"
    process "MERGE_FASTQ"

    test("Should run without failures") {

        when {
            params {
                min_read_length = 1000
                min_qscore = 9
            }
            process {
                """
                input[0] = Channel.from("$baseDir/tests/input/merge_fastq/")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            
        }

    }

    test("Should output empty file") {

        when {
            params {
                min_read_length = 10000
                min_qscore = 9
            }
            process {
                """
                input[0] = Channel.from("$baseDir/tests/input/merge_fastq/").view()
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
