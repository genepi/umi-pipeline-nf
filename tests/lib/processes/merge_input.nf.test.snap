{
    "Should output empty file": {
        "content": [
            {
                "0": [
                    [
                        "merge_fastq",
                        "target",
                        "merged.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "merged_fastq": [
                    [
                        "merge_fastq",
                        "target",
                        "merged.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-12T11:18:40.366759"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "merge_fastq",
                        "target",
                        "merged.fastq:md5,226ecd48aa7c4b401dcbb0a3fe771365"
                    ]
                ],
                "merged_fastq": [
                    [
                        "merge_fastq",
                        "target",
                        "merged.fastq:md5,226ecd48aa7c4b401dcbb0a3fe771365"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-12T11:17:41.974206"
    }
}